packages/db/schema.prisma

```diff
@@ -419,3 +431,26 @@ model BetaTokens {
 
   @@index([userId])
 }
+
+model Notification {
+  id               Int              @id @default(autoincrement())
+  createdAt        DateTime         @default(now())
+  updatedAt        DateTime         @updatedAt
+  type             NotificationType
+  toUser           User             @relation("notificationToUser", fields: [toUserId], references: [id])
+  toUserId         String
+  fromUser         User             @relation("notificationFromUser", fields: [fromUserId], references: [id])
+  fromUserId       String
+  sharedSolutionId Int?
+  sharedSolution   SharedSolution?  @relation(fields: [sharedSolutionId], references: [id])
```

Comment:
Instead of adding multiple resource-specific columns to the table, such as `comment`, `sharedSolution`, etc., could we use a more generic approach with something like `objectType` and `objectId`? This would allow notifications to be associated with any type of object. Given that we maintain data integrity at the application level, this approach would simplify the schema and accommodate various notification types.

Reply:
prisma doesn't allow for polymorphic relationships. i'd rather not lose the strong typing tbh

Reply:
Understood!